File: 프로그래머스/unrated/181918. 배열 만들기 4/배열 만들기 4.py
def solution(arr):
    stk = []    
    i = 0
    while True:
        if i >= len(arr):
            return stk
        if not stk:
            stk.append(arr[i])
            i += 1
        elif stk[-1] < arr[i]:
            stk.append(arr[i])
            i += 1
        else:
            stk.pop()
    
        

File: 프로그래머스/0/181918. 배열 만들기 4/배열 만들기 4.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        Stack<Integer> stk = new Stack<>();
        int i = 0;
        while (i < arr.length) {
            if (stk.isEmpty()) {
                stk.push(arr[i]);
                i++;
            } else if(stk.peek() < arr[i]) {
                stk.push(arr[i]);
                i++;
            } else {
                stk.pop();
            }
        }
        return stk.stream().mapToInt(v -> v).toArray();
    }
}